File: app/ClientApp/my-first-crud/src/models/Product.ts
export interface Product {
  productId: number
  name: string
  productCategory: productCategory
  description: string
  price: number
  sku: string
  code: string
}

enum productCategory {
  Clothing = 1,
  Footware = 2,
  Electronics = 3,
  Household = 4
}

File: app/ClientApp/clientApi/WebApi.Domain/Product.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace WebApi.Qumulus.Domain
{
    public class Product
    {
        public int ProductId { get; set; }
        public string? Name { get; set; }
        public ProductCategory ProductCategory { get; set; }

        [Column(TypeName = "nvarch(100)")]
        public string? Description { get; set; }
        public decimal Price { get; set; }
        public string? SKU { get; set; }
        public string? Code { get; set; }
    }
}